pallets/msa/src/migration.rs

```diff
@@ -0,0 +1,158 @@
+//! Migrations for the MSA Pallet
+
+use super::*;
+use frame_support::storage_alias;
+
+/// Data Structures that were removed
+pub mod v0 {
+	use super::*;
+
+	/// Replaced by `PayloadSignatureRegistryRingPointer`
+	#[storage_alias]
+	pub(super) type PayloadSignatureBucketCount<T: Config> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		u64, // bucket number
+		u32, // number of signatures
+		ValueQuery,
+	>;
+
+	/// Replaced with `PayloadSignatureRegistryRing`
+	#[storage_alias]
+	pub(super) type PayloadSignatureRegistry<T: Config> = StorageDoubleMap<
+		Pallet<T>,      // prefix
+		Twox64Concat,   // hasher for key1
+		u64,            // Bucket number. Stored as BlockNumber because I'm done arguing with rust about it.
+		Twox64Concat,   // hasher for key2
+		MultiSignature, // An externally-created Signature for an external payload, provided by an extrinsic
+		u64,            // An actual flipping block number.
+	>;
+}
+
+/// MSA Pallet Migration Triggers
+pub fn migrate<T: Config>() -> Weight {
+	let version = StorageVersion::get::<Pallet<T>>();
+	let mut weight: Weight = Weight::zero();
+
+	if version < 1 {
+		weight = weight.saturating_add(v1::migrate::<T>());
+		// Updated version inside so we can drain the prefix if needed
+	}
+
+	weight
+}
+
+/// Migrating to remove old storage
```

Comment:
This does not try to deal with the situation of moving the old signatures into the new system. This is due to the system not being used on mainnet yet.